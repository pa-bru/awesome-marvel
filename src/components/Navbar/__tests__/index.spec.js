import React from 'react'
import { shallow } from 'enzyme'
import { Navbar } from '../'

import IconButton from '@material-ui/core/IconButton'

describe('Navbar', () => {
  it('should render correctly', () => {
    // When
    const tree = shallow(<Navbar classes={{}} />)

    // Then
    expect(tree).toMatchSnapshot()
  })

  it('should render with a child', () => {
    // Given
    const Child = () => <div>COUCOU</div>

    // When
    const tree = shallow(
      <Navbar classes={{}}>
        <Child />
      </Navbar>
    )

    // Then
    expect(tree).toMatchSnapshot()
  })

  it('should toggle the menu', () => {
    // Given
    const tree = shallow(<Navbar classes={{}} />)
    const initial = tree.state('isMenuOpen')
    const button = tree.find(IconButton).first()

    // When
    button.simulate('click')

    // Then
    expect(tree.state('isMenuOpen')).toEqual(!initial)
  })
})
