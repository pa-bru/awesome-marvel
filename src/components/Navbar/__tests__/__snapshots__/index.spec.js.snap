// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render correctly 1`] = `
<React.Fragment>
  <WithStyles(AppBar)
    position="fixed"
  >
    <WithStyles(Toolbar)>
      <WithStyles(IconButton)
        aria-label="Menu"
        color="inherit"
        onClick={[Function]}
      >
        <pure(Menu) />
      </WithStyles(IconButton)>
      <WithStyles(Typography)
        color="inherit"
        variant="title"
      >
        Awesome Marvel
      </WithStyles(Typography)>
    </WithStyles(Toolbar)>
  </WithStyles(AppBar)>
  <WithStyles(Drawer)
    onClose={[Function]}
    open={false}
  >
    <div
      onClick={[Function]}
      role="button"
    >
      <WithStyles(List)
        component="nav"
      >
        <Link
          replace={false}
          to="/"
        >
          <WithStyles(ListItem)
            button={true}
          >
            <WithStyles(ListItemIcon)>
              <pure(Home) />
            </WithStyles(ListItemIcon)>
            <WithStyles(ListItemText)
              primary="Home"
            />
          </WithStyles(ListItem)>
        </Link>
        <Link
          replace={false}
          to="/about"
        >
          <WithStyles(ListItem)
            button={true}
          >
            <WithStyles(ListItemIcon)>
              <pure(Info) />
            </WithStyles(ListItemIcon)>
            <WithStyles(ListItemText)
              primary="About"
            />
          </WithStyles(ListItem)>
        </Link>
      </WithStyles(List)>
      <WithStyles(Divider) />
      <WithStyles(Connect(HistoryList)) />
    </div>
  </WithStyles(Drawer)>
</React.Fragment>
`;

exports[`Navbar should render with a child 1`] = `
<React.Fragment>
  <WithStyles(AppBar)
    position="fixed"
  >
    <WithStyles(Toolbar)>
      <WithStyles(IconButton)
        aria-label="Menu"
        color="inherit"
        onClick={[Function]}
      >
        <pure(Menu) />
      </WithStyles(IconButton)>
      <WithStyles(Typography)
        color="inherit"
        variant="title"
      >
        Awesome Marvel
      </WithStyles(Typography)>
    </WithStyles(Toolbar)>
  </WithStyles(AppBar)>
  <WithStyles(Drawer)
    onClose={[Function]}
    open={false}
  >
    <div
      onClick={[Function]}
      role="button"
    >
      <WithStyles(List)
        component="nav"
      >
        <Link
          replace={false}
          to="/"
        >
          <WithStyles(ListItem)
            button={true}
          >
            <WithStyles(ListItemIcon)>
              <pure(Home) />
            </WithStyles(ListItemIcon)>
            <WithStyles(ListItemText)
              primary="Home"
            />
          </WithStyles(ListItem)>
        </Link>
        <Link
          replace={false}
          to="/about"
        >
          <WithStyles(ListItem)
            button={true}
          >
            <WithStyles(ListItemIcon)>
              <pure(Info) />
            </WithStyles(ListItemIcon)>
            <WithStyles(ListItemText)
              primary="About"
            />
          </WithStyles(ListItem)>
        </Link>
      </WithStyles(List)>
      <WithStyles(Divider) />
      <WithStyles(Connect(HistoryList)) />
    </div>
  </WithStyles(Drawer)>
  <Child />
</React.Fragment>
`;
